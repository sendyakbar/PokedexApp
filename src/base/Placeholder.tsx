import React from 'react';
import {StyleSheet, View, ViewStyle} from 'react-native';

type Props = ViewStyle;

export default function Placeholder(props: Props): JSX.Element {
  const styles = styleGenerator(props);
  return <View style={styles.container} />;
}

type StyleType = {
  container: ViewStyle;
};

const styleGenerator = (payload: ViewStyle) =>
  StyleSheet.create<StyleType>({
    container: {
      ...payload,
      backgroundColor: '#EEEEEE',
      borderRadius: 4,
    },
  });
