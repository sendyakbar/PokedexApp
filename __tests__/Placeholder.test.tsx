import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

import Placeholder from '../src/base/Placeholder';

it('renders correctly', () => {
  const tree = renderer
    .create(<Placeholder height={55} width="100%" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
