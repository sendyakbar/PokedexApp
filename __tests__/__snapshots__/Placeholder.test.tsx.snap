// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#EEEEEE",
      "borderRadius": 4,
      "height": 55,
      "width": "100%",
    }
  }
/>
`;
